src/tensors/levels/sparse_list_levels.jl

```diff
@@ -141,8 +141,8 @@ end
 
 is_level_injective(ctx, lvl::VirtualSparseListLevel) = [is_level_injective(ctx, lvl.lvl)..., false]
 function is_level_atomic(ctx, lvl::VirtualSparseListLevel)
-    (below, atomic) = is_level_atomic(ctx, lvl.lvl)
-    return ([below; [atomic]], atomic)
+    (below, Mutex) = is_level_atomic(ctx, lvl.lvl)
```

Comment:
I forget how this works, but shouldn't a false be being added somewhere here?

Reply:
and in many of these.

Reply:
I honestly don't know why this changed.

Reply:
This is how it was before we started. Perhaps I need to make an issue for this. I'll look into it. One related question I had for you: This PR delays the lowering of atomics until just before their child is unfurled or accessed. That doesn't change this analysis right?

Reply:
That is going to require some thought for me to answer. Hmm.